vsintegration/src/FSharp.Editor/Completion/CompletionProvider.fs

```diff
@@ -52,85 +57,99 @@ type internal FSharpCompletionProvider
-        async {
-            let! parseResults, checkFileAnswer = checker.ParseAndCheckFileInProject(filePath, textVersionHash, sourceText.ToString(), options)
-            match parseResults.ParseTree, checkFileAnswer with
-            | _, FSharpCheckFileAnswer.Aborted
-            | None, _ -> return None
-            | Some parsedInput, FSharpCheckFileAnswer.Succeeded(checkFileResults) ->
-                let textLines = sourceText.Lines
-                let caretLinePos = textLines.GetLinePosition(caretPosition)
-                let entityKind = UntypedParseImpl.GetEntityKind(Pos.fromZ caretLinePos.Line caretLinePos.Character, parsedInput)
-                
-                let caretLine = textLines.GetLineFromPosition(caretPosition)
-                let fcsCaretLineNumber = Line.fromZ caretLinePos.Line  // Roslyn line numbers are zero-based, FSharp.Compiler.Service line numbers are 1-based
-                let caretLineColumn = caretLinePos.Character
-                
-                let qualifyingNames, partialName = QuickParse.GetPartialLongNameEx(caretLine.ToString(), caretLineColumn - 1) 
-                let! declarations = checkFileResults.GetDeclarationListInfo(Some(parseResults), fcsCaretLineNumber, caretLineColumn, caretLine.ToString(), qualifyingNames, partialName)
-                
-                let results = List<CompletionItem>()
-                
-                for declarationItem in declarations.Items do
-                    let glyph = CommonRoslynHelpers.FSharpGlyphToRoslynGlyph declarationItem.GlyphMajor
-                    let name =
-                        match entityKind with
-                        | Some EntityKind.Attribute when declarationItem.IsAttribute && declarationItem.Name.EndsWith "Attribute"  ->
-                            declarationItem.Name.[0..declarationItem.Name.Length - attributeSuffixLength - 1] 
-                        | _ -> declarationItem.Name
-                    let completionItem = CommonCompletionItem.Create(name, glyph = Nullable glyph)
-                    
-                    let completionItem =
-                        if declarationItem.Name <> declarationItem.NameInCode then
-                            completionItem.AddProperty(NameInCodePropName, declarationItem.NameInCode)
-                        else completionItem
-
-                    declarationItemsCache.Remove(completionItem.DisplayText) |> ignore // clear out stale entries if they exist
-                    declarationItemsCache.Add(completionItem.DisplayText, declarationItem)
-                    results.Add(completionItem)
+        asyncMaybe {
+            let! parseResults, parsedInput, checkFileResults = checker.ParseAndCheckDocument(filePath, textVersionHash, sourceText.ToString(), options)
+
+            let textLines = sourceText.Lines
```

Comment:
Again may be best to minimize the diff with some non-standard indentation at least for this PR